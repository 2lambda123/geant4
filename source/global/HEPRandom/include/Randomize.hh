// This code implementation is the intellectual property of
// the GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: Randomize.hh,v 1.3 2000-08-02 10:48:21 gcosmo Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
#ifndef randomize_h
#define randomize_h 1

#include <CLHEP/Random/Randomize.h>

// Additional distributions ...
//
#include <CLHEP/Random/RandGaussQ.h>
#include <CLHEP/Random/RandGaussT.h>
#include <CLHEP/Random/RandPoissonQ.h>
#include <CLHEP/Random/RandPoissonT.h>
#include <CLHEP/Random/RandLandau.h>
#include <CLHEP/Random/RandBit.h>

#define G4UniformRand() HepRandom::getTheEngine()->flat()

#endif // randomize_h 
