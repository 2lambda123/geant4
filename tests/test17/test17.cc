//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: test17.cc,v 1.10 2005-05-20 09:29:13 vnivanch Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
// 
// --------------------------------------------------------------
//      GEANT 4 - test17
//
// --------------------------------------------------------------
// Comments
//
// 18.08.2000 V.Ivanchenko clean up visualisation and dummy output
//
// --------------------------------------------------------------

#include "G4RunManager.hh"
#include "G4UImanager.hh"
#include "G4UIterminal.hh"
#include "Randomize.hh"

#include "Test17DetectorConstruction.hh"
#include "Test17PhysicsList.hh"
#include "Test17PrimaryGeneratorAction.hh"
#include "Test17RunAction.hh"
#include "Test17EventAction.hh"
#include "Test17SteppingAction.hh"

int main(int argc,char** argv) {

  //choose the Random engine
  HepRandom::setTheEngine(new RanecuEngine);
  
  // Construct the default run manager
  G4RunManager * runManager = new G4RunManager;

  // set mandatory initialization classes
  Test17DetectorConstruction* detector;
  detector = new Test17DetectorConstruction;
  runManager->SetUserInitialization(detector);
  runManager->SetUserInitialization(new Test17PhysicsList(detector));

  // set user action classes
  runManager->SetUserAction(new Test17PrimaryGeneratorAction(detector));
  Test17RunAction* runaction = new Test17RunAction;
  runManager->SetUserAction(runaction);

  Test17EventAction* eventaction = new Test17EventAction(runaction);
  runManager->SetUserAction(eventaction);

  Test17SteppingAction* steppingaction = new Test17SteppingAction(detector,
                                               eventaction, runaction);
  runManager->SetUserAction(steppingaction);

  //Initialize G4 kernel
//  runManager->Initialize();

  // get the pointer to the User Interface manager
    G4UImanager* UI = G4UImanager::GetUIpointer();

  if (argc==1)   // Define UI terminal for interactive mode
    {
     G4UIsession * session = new G4UIterminal;
     UI->ApplyCommand("/control/execute init.mac");
     session->SessionStart();
     delete session;
    }
  else           // Batch mode
    {
     G4String command = "/control/execute ";
     G4String fileName = argv[1];
     UI->ApplyCommand(command+fileName);
    }

  // job termination
  delete runManager;

  return 0;
}

