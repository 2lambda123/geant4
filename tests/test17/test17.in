#
# Macro file for the initialization phase of "test17" based on "TestEm6.cc"
#
# It creates the default geometry (simple absorber box )        
#
/control/verbose 2
/event/setverbose 0
#
/run/initialize
#
/gun/particle proton
#/gun/zvertex -0.16 mum
/gun/energy 500 keV
/tracking/verbose 0
/run/beamOn 1000
#
/gun/particle anti_proton
/run/beamOn 1000

#scaled energy
/gun/particle IonC12
/gun/energy 6000 keV
/run/beamOn 1000

