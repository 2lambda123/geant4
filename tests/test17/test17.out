
*************************************************************
 Geant4 version $Name: not supported by cvs2svn $   (25-March-2004)
                      Copyright : Geant4 Collaboration
                      Reference : NIM A 506 (2003), 250-303.
                            WWW : http://cern.ch/geant4
*************************************************************

G4VUserPhysicsList::SetVerboseLevel  : Verbose level is set to 2
Material available: Beryllium
Material available: Aluminum
Material available: Carbon
Material available: Silicon
Material available: liquidArgon
Material available: liquidXenon
Material available: Iron
Material available: Copper
Material available: Gold
Material available: Lead
Material available: Water
Material available: Methane
Material available: Graphite

***** Table : Nb of materials = 14 *****

 Material: Beryllium     density:  1.848 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:  35.267 cm 
   --->  Element: Beryllium      Z =  4.0   N =   9.0   A =   9.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Aluminum     density:  2.700 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:   8.893 cm 
   --->  Element: Aluminum      Z = 13.0   N =  27.0   A =  26.98 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:   Carbon     density:  2.000 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:  21.348 cm 
   --->  Element:   Carbon      Z =  6.0   N =  12.0   A =  12.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:  Silicon     density:  2.330 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:   9.368 cm 
   --->  Element:  Silicon      Z = 14.0   N =  28.1   A =  28.09 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: liquidArgon     density:  1.390 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:  14.065 cm 
   --->  Element: liquidArgon      Z = 18.0   N =  40.0   A =  39.95 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: liquidXenon     density:  3.020 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:   2.808 cm 
   --->  Element: liquidXenon      Z = 54.0   N = 131.3   A = 131.29 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:     Iron     density:  7.870 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.759 cm 
   --->  Element:     Iron      Z = 26.0   N =  55.9   A =  55.85 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:   Copper     density:  8.960 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.436 cm 
   --->  Element:   Copper      Z = 29.0   N =  63.5   A =  63.55 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:     Gold     density: 19.320 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:   3.344 mm 
   --->  Element:     Gold      Z = 79.0   N = 197.0   A = 196.97 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:     Lead     density: 11.350 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:   5.612 mm 
   --->  Element:     Lead      Z = 82.0   N = 207.2   A = 207.19 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:    Water H_2O   density:  1.000 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:  36.092 cm 
   --->  Element: Hydrogen  H   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  11.21 %  Abundance  66.67 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  88.79 %  Abundance  33.33 %

 Material:  Methane CH_4   density:  0.667 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 696.375 m  
   --->  Element: Hydrogen  H   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  25.19 %  Abundance  80.00 %
   --->  Element:   Carbon  C   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass:  74.81 %  Abundance  20.00 %

 Material: Graphite Graphite   density:  2.265 g/cm3   temperature: 273.15 K  pressure:   1.00 atm  RadLength:  18.834 cm 
   --->  Element:   Carbon  C   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material:      Air     density:  1.000 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 367.858 m  
   --->  Element: Nitrogen  N   Z =  7.0   N =  14.0   A =  14.01 g/mole  fractionMass:  70.00 %  Abundance  72.71 %
   --->  Element:   Oxygen  O   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  30.00 %  Abundance  27.29 %



 The  WORLD   is made of 21 cm  of Air. The transverse size (YZ) of the world is 12 cm 
 The ABSORBER is made of 1 items of 10 cm  of Water. The transverse size (YZ) is 10 cm 
 X position of the middle of the absorber 0 fm 
G4VUserPhysicsList::Construct()
Construct particles 
Construct processes 
Ionic processes for GenericIon
Ionic processes for IonC12
Ionic processes for alpha
Hadronic processes for anti_proton
Ionic processes for deuteron

phot:  Total cross sections from Sandia parametrisation. 
Hadronic processes for kaon+
Hadronic processes for kaon-
Hadronic processes for pi+
Hadronic processes for pi-
Hadronic processes for proton
Test17PhysicsList::SetCuts:CutLength : 100 mum
Set cuts for all particles! 
/run/verbose 1
/test17/cutG 10 micrometer
/test17/cutE 10 micrometer
/run/initialize
#
/run/initialize
#
/gun/particle proton
/gun/energy 500 keV
/tracking/verbose 0
/run/beamOn 1000

compt:  Total cross sections from a parametrisation. Good description from 10 KeV to (100/Z) GeV. 
       Scattered gamma energy according Klein-Nishina.
        PhysicsTables from 1 keV to 100 GeV in 80 bins. 

conv:  Total cross sections from a parametrisation. Good description from 1.5 MeV to 100 GeV for all Z. 
        e+e- energies according Bethe-Heitler
        PhysicsTables from 1.022 MeV to 100 GeV in 100 bins. 

msc:  Model variant of multiple scattering for e-
      Lambda tables from 100 eV  to 100 TeV in 120 bins.
      Boundary algorithm is active with facrange= 0.199

eIoni:   tables are built for  e-
      dE/dx and range tables from 100 eV  to 100 TeV in 120 bins.
      Lambda tables from threshold to 100 TeV in 120 bins.
      Step function: finalRange(mm)= 1, dRoverRange= 1, integral: 1
      Delta cross sections from Moller+Bhabha, good description from 1 KeV to 100 GeV.

eBrem:   tables are built for  e-
      dE/dx and range tables from 100 eV  to 100 TeV in 120 bins.
      Lambda tables from threshold to 100 TeV in 120 bins.
      Total cross sections from a parametrisation based on the EEDL data library. 
      Good description from 1 KeV to 100 GeV, log scale extrapolation above 100 GeV.

eIoni:   tables are built for  e+
      dE/dx and range tables from 100 eV  to 100 TeV in 120 bins.
      Lambda tables from threshold to 100 TeV in 120 bins.
      Step function: finalRange(mm)= 1, dRoverRange= 1, integral: 1
      Delta cross sections from Moller+Bhabha, good description from 1 KeV to 100 GeV.

eBrem:   tables are built for  e+
      dE/dx and range tables from 100 eV  to 100 TeV in 120 bins.
      Lambda tables from threshold to 100 TeV in 120 bins.
      Total cross sections from a parametrisation based on the EEDL data library. 
      Good description from 1 KeV to 100 GeV, log scale extrapolation above 100 GeV.

annihil:  Total cross section from Heilter formula(annihilation into 2 photons).
        gamma energies sampled according Heitler
        PhysicsTables from 10 keV to 10 TeV in 100 bins. 

msc:  Model variant of multiple scattering for proton
      Lambda tables from 100 eV  to 100 TeV in 120 bins.
      Boundary algorithm is active with facrange= 0.199
G4AugerData for Element no. 6 are loaded
G4AugerData for Element no. 7 are loaded
G4AugerData for Element no. 8 are loaded
G4AugerData for Element no. 13 are loaded
G4AugerData for Element no. 14 are loaded
G4AugerData for Element no. 18 are loaded
G4AugerData for Element no. 26 are loaded
G4AugerData for Element no. 29 are loaded
G4AugerData for Element no. 54 are loaded
G4AugerData for Element no. 79 are loaded
G4AugerData for Element no. 82 are loaded
AugerTransitionTable complete

hLowEIoni:    Knock-on electron cross sections . 
        Good description above the mean excitation energy.
        Delta ray energy sampled from  differential Xsection.
        PhysicsTables from 10 eV  to 0.1 TeV  in 360 bins.
        Electronic stopping power model is  ICRU_R49p
        from 1 keV  to 2 MeV .

        Parametrisation model for antiprotons is  ICRU_R49p
        from 25 keV  to 2 MeV .
        Parametrization of the Barkas effect is switched on.
        Nuclear stopping power model is ICRU_R49
G4hLowEnergyIonisation::BuildPhysicsTable for GenericIon mass(MeV)= 938.272 charge= 1 type= nucleus
Cuts are defined 
G4hLowEnergyIonisation::BuildPhysicsTable: Loss table is built 
G4hLowEnergyIonisation::BuildPhysicsTable: DEDX table will be built 
G4hLowEnergyIonisation::BuildPhysicsTable: end for GenericIon
G4hLowEnergyIonisation::BuildPhysicsTable for IonC12 mass(MeV)= 11149.5 charge= 6 type= static_nucleus
Cuts are defined 
G4hLowEnergyIonisation::BuildPhysicsTable: Loss table is built 
G4hLowEnergyIonisation::BuildPhysicsTable: DEDX table will be built 
G4hLowEnergyIonisation::BuildPhysicsTable: end for IonC12
G4hLowEnergyIonisation::BuildPhysicsTable for alpha mass(MeV)= 3727.42 charge= 2 type= nucleus
Cuts are defined 
G4hLowEnergyIonisation::BuildPhysicsTable: Loss table is built 
G4hLowEnergyIonisation::BuildPhysicsTable: DEDX table will be built 
G4hLowEnergyIonisation::BuildPhysicsTable: end for alpha
G4hLowEnergyIonisation::BuildPhysicsTable for deuteron mass(MeV)= 1875.61 charge= 1 type= nucleus
Cuts are defined 
G4hLowEnergyIonisation::BuildPhysicsTable: Loss table is built 
G4hLowEnergyIonisation::BuildPhysicsTable: DEDX table will be built 
G4hLowEnergyIonisation::BuildPhysicsTable: end for deuteron

msc:  Model variant of multiple scattering for mu+
      Lambda tables from 100 eV  to 100 TeV in 120 bins.
      Boundary algorithm is active with facrange= 0.199

MuIoni:   tables are built for  mu+
      dE/dx and range tables from 100 eV  to 100 TeV in 120 bins.
      Lambda tables from threshold to 100 TeV in 120 bins.
      Step function: finalRange(mm)= 1, dRoverRange= 0.2, integral: 0
      Bether-Bloch model for E > 0.2 MeV, parametrisation of Bragg peak below, 
      radiative corrections for E > 1 GeV

MuBrems:   tables are built for  mu+
      dE/dx and range tables from 100 eV  to 100 TeV in 120 bins.
      Lambda tables from threshold to 100 TeV in 120 bins.
      Parametrised model 

MuPairProd:   tables are built for  mu+
      dE/dx and range tables from 100 eV  to 100 TeV in 120 bins.
      Lambda tables from threshold to 100 TeV in 120 bins.
      Parametrised model 
### All dEdx and Range tables are built #####

MuIoni:   tables are built for  mu-
      dE/dx and range tables from 100 eV  to 100 TeV in 120 bins.
      Lambda tables from threshold to 100 TeV in 120 bins.
      Step function: finalRange(mm)= 1, dRoverRange= 0.2, integral: 0
      Bether-Bloch model for E > 0.2 MeV, parametrisation of Bragg peak below, 
      radiative corrections for E > 1 GeV

msc:  Model variant of multiple scattering for pi-
      Lambda tables from 100 eV  to 100 TeV in 120 bins.
      Boundary algorithm is active with facrange= 0.199

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes     recalculation needed : No 
 Material : Air
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1 eV     e- 4.01595 eV     e+ 4.01595 eV 
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 1     used in the geometry : Yes     recalculation needed : No 
 Material : Water
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1.07536 keV    e- 84.6034 keV    e+ 83.5687 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

====================================================================

### Run 0 start.
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=2.32s Real=4.27s Sys=1.94s
 ================== run summary =====================

 end of Run TotNbofEvents = 1000 for proton with Ekin = 0.5 MeV
    Track Length in absorber = 9.1806 +- 0.00543904  microns 
    CSDA  Range  in absorber = 9.05329 +- 0.00579619  microns 

    Energy deposit in absorber = 0.5  MeV 

### Comparison with the ICRU49 data: 
    Track Length (G4 - ICRU49) = 0.121599 +- 0.00543904 microns 
    CSDA  Range  (G4 - ICRU49) = 0.184293 +- 0.00579619 microns 

Average Number electrons per event = 0
Average Number photons per event   = 0

#
/gun/particle anti_proton
/run/beamOn 1000

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes     recalculation needed : No 
 Material : Air
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1 eV     e- 4.01595 eV     e+ 4.01595 eV 
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 1     used in the geometry : Yes     recalculation needed : No 
 Material : Water
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1.07536 keV    e- 84.6034 keV    e+ 83.5687 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

====================================================================

### Run 1 start.
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=2.3s Real=4.22s Sys=1.91s
 ================== run summary =====================

 end of Run TotNbofEvents = 1000 for anti_proton with Ekin = 0.5 MeV
    Track Length in absorber = 9.59149 +- 0.00629657  microns 
    CSDA  Range  in absorber = 9.46386 +- 0.00612297  microns 

    Energy deposit in absorber = 0.5  MeV 

Average Number electrons per event = 0
Average Number photons per event   = 0

#scaled energy
/gun/particle IonC12
/gun/energy 6000 keV
/run/beamOn 1000

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes     recalculation needed : No 
 Material : Air
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1 eV     e- 4.01595 eV     e+ 4.01595 eV 
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 1     used in the geometry : Yes     recalculation needed : No 
 Material : Water
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1.07536 keV    e- 84.6034 keV    e+ 83.5687 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

====================================================================

### Run 2 start.
Start Run processing.
EvtNo= 3  length= 0.00843332  xend= 0.00382408
EvtNo= 16  length= 0.00844533  xend= 0.0036791
EvtNo= 41  length= 0.00839609  xend= 0.000907912
EvtNo= 44  length= 0.00851927  xend= 0.000600708
EvtNo= 46  length= 0.00846101  xend= 0.000147361
EvtNo= 49  length= 0.00840546  xend= 0.00251327
EvtNo= 61  length= 0.00842454  xend= 0.00041738
EvtNo= 66  length= 0.00849077  xend= 0.0019715
EvtNo= 81  length= 0.00844201  xend= 0.000883018
EvtNo= 84  length= 0.00838181  xend= 0.00186301
EvtNo= 85  length= 0.00843048  xend= 0.000816792
EvtNo= 86  length= 0.00843624  xend= 0.00159236
EvtNo= 94  length= 0.00837822  xend= 0.00106884
EvtNo= 103  length= 0.00843501  xend= 0.00404965
EvtNo= 104  length= 0.00845872  xend= 0.0029785
EvtNo= 106  length= 0.00844722  xend= 0.00105453
EvtNo= 110  length= 0.00849706  xend= 0.00382759
EvtNo= 121  length= 0.00847681  xend= 0.00141948
EvtNo= 124  length= 0.00846159  xend= 0.00282853
EvtNo= 145  length= 0.00845702  xend= 0.00370642
EvtNo= 165  length= 0.00847944  xend= 0.00391452
EvtNo= 167  length= 0.00854648  xend= 0.00225607
EvtNo= 184  length= 0.00836932  xend= 0.000677807
EvtNo= 186  length= 0.00849416  xend= 0.00111306
EvtNo= 192  length= 0.00845366  xend= 0.00174459
EvtNo= 194  length= 0.00845695  xend= 0.00138982
EvtNo= 214  length= 0.00846272  xend= 0.000862556
EvtNo= 215  length= 0.00845529  xend= 0.00337902
EvtNo= 223  length= 0.00848602  xend= 0.000337968
EvtNo= 228  length= 0.00853243  xend= 0.00305388
EvtNo= 235  length= 0.00846858  xend= 0.00066714
EvtNo= 240  length= 0.00840523  xend= 0.00361028
EvtNo= 256  length= 0.00849096  xend= 0.00355393
EvtNo= 279  length= 0.00843118  xend= 0.00200196
EvtNo= 280  length= 0.00846868  xend= 0.000941232
EvtNo= 282  length= 0.00847792  xend= 0.00237627
EvtNo= 293  length= 0.0084179  xend= 6.45285e-05
EvtNo= 294  length= 0.00849503  xend= 0.00398798
EvtNo= 297  length= 0.00840131  xend= 0.0010398
EvtNo= 305  length= 0.00839297  xend= 0.00412448
EvtNo= 310  length= 0.0084096  xend= 0.00358813
EvtNo= 313  length= 0.00848707  xend= 0.00283975
EvtNo= 316  length= 0.00842373  xend= 0.00263695
EvtNo= 321  length= 0.00848442  xend= 0.0033975
EvtNo= 324  length= 0.00846445  xend= 0.00335734
EvtNo= 325  length= 0.00844591  xend= 0.00199649
EvtNo= 330  length= 0.00838947  xend= 0.00376399
EvtNo= 333  length= 0.00847069  xend= 0.000911832
EvtNo= 345  length= 0.00847118  xend= 0.00237245
EvtNo= 352  length= 0.00843274  xend= 0.00179151
EvtNo= 361  length= 0.008446  xend= 0.00158733
EvtNo= 362  length= 0.0084026  xend= 0.00205556
EvtNo= 363  length= 0.00843278  xend= 0.00405011
EvtNo= 368  length= 0.00842684  xend= 0.00132255
EvtNo= 376  length= 0.00843375  xend= 0.00039787
EvtNo= 383  length= 0.00848479  xend= 0.00182566
EvtNo= 385  length= 0.00849889  xend= 0.000372741
EvtNo= 391  length= 0.00840872  xend= 0.00327462
EvtNo= 395  length= 0.00840208  xend= 0.00230395
EvtNo= 403  length= 0.00846691  xend= 0.00365622
EvtNo= 412  length= 0.00850007  xend= 0.00250242
EvtNo= 414  length= 0.0084655  xend= 0.00220079
EvtNo= 455  length= 0.00843626  xend= 0.0031783
EvtNo= 457  length= 0.00851183  xend= 0.00074998
EvtNo= 458  length= 0.00841445  xend= 0.00274316
EvtNo= 461  length= 0.00839281  xend= 0.0035488
EvtNo= 467  length= 0.00852903  xend= 0.00325751
EvtNo= 469  length= 0.00849654  xend= 0.00242409
EvtNo= 477  length= 0.00852067  xend= 0.00207624
EvtNo= 484  length= 0.00849476  xend= 0.00148218
EvtNo= 496  length= 0.00848325  xend= 0.00174019
EvtNo= 501  length= 0.00846766  xend= 0.00204421
EvtNo= 522  length= 0.0084358  xend= 0.00416891
EvtNo= 528  length= 0.00841887  xend= 0.00412901
EvtNo= 530  length= 0.0083657  xend= 0.00119693
EvtNo= 536  length= 0.00845175  xend= 0.00407619
EvtNo= 541  length= 0.00844377  xend= 0.00289633
EvtNo= 545  length= 0.00845861  xend= 9.91159e-05
EvtNo= 557  length= 0.00851557  xend= 0.0014456
EvtNo= 577  length= 0.00845701  xend= 0.00326515
EvtNo= 578  length= 0.00847464  xend= 0.00337336
EvtNo= 582  length= 0.00841644  xend= 0.00412034
EvtNo= 587  length= 0.00837812  xend= 0.00277563
EvtNo= 594  length= 0.00851441  xend= 0.00309956
EvtNo= 598  length= 0.00847999  xend= 0.00246886
EvtNo= 617  length= 0.00842369  xend= 0.00179032
EvtNo= 619  length= 0.00851707  xend= 0.00354751
EvtNo= 634  length= 0.00838735  xend= 0.00192127
EvtNo= 637  length= 0.00842826  xend= 0.00206811
EvtNo= 648  length= 0.0084011  xend= 0.00279876
EvtNo= 649  length= 0.00847018  xend= 0.00342575
EvtNo= 658  length= 0.00844403  xend= 0.000370892
EvtNo= 660  length= 0.00848816  xend= 0.00168133
EvtNo= 663  length= 0.00840642  xend= 0.00314907
EvtNo= 672  length= 0.00836899  xend= 0.00257117
EvtNo= 674  length= 0.00835297  xend= 0.00406043
EvtNo= 685  length= 0.00851567  xend= 0.00265201
EvtNo= 687  length= 0.00848068  xend= 0.00163697
EvtNo= 689  length= 0.00844611  xend= 0.00287824
EvtNo= 696  length= 0.00851638  xend= 0.00145016
EvtNo= 698  length= 0.00841162  xend= 0.00381827
EvtNo= 705  length= 0.00846007  xend= 9.92629e-05
EvtNo= 709  length= 0.00851109  xend= 0.00199266
EvtNo= 719  length= 0.00850207  xend= 0.0010646
EvtNo= 721  length= 0.00843325  xend= 0.000493692
EvtNo= 724  length= 0.00842958  xend= 0.00256042
EvtNo= 726  length= 0.0084823  xend= 0.00135426
EvtNo= 727  length= 0.00841603  xend= 0.00364334
EvtNo= 740  length= 0.00849195  xend= 8.74312e-05
EvtNo= 746  length= 0.00845351  xend= 0.00390241
EvtNo= 755  length= 0.00848832  xend= 0.00378421
EvtNo= 758  length= 0.00849263  xend= 0.00232939
EvtNo= 759  length= 0.00843812  xend= 0.00385142
EvtNo= 770  length= 0.00836338  xend= 0.00120029
EvtNo= 772  length= 0.00847012  xend= 0.00314578
EvtNo= 781  length= 0.00844417  xend= 0.00231655
EvtNo= 783  length= 0.00845617  xend= 0.00307388
EvtNo= 792  length= 0.00840592  xend= 0.00146394
EvtNo= 803  length= 0.00851866  xend= 0.00230992
EvtNo= 807  length= 0.00844801  xend= 0.0024688
EvtNo= 814  length= 0.00841397  xend= 0.00404549
EvtNo= 817  length= 0.00847917  xend= 0.00344189
EvtNo= 829  length= 0.0084857  xend= 0.00294499
EvtNo= 835  length= 0.00843954  xend= 0.00367813
EvtNo= 882  length= 0.00848451  xend= 0.00389019
EvtNo= 885  length= 0.00842704  xend= 0.000134334
EvtNo= 893  length= 0.00844131  xend= 0.00220703
EvtNo= 896  length= 0.00848907  xend= 0.00311222
EvtNo= 899  length= 0.00843393  xend= 0.00234512
EvtNo= 902  length= 0.00845634  xend= 0.00255386
EvtNo= 903  length= 0.00840659  xend= 0.00249903
EvtNo= 907  length= 0.00846897  xend= 0.00313485
EvtNo= 928  length= 0.00844264  xend= 0.00237203
EvtNo= 930  length= 0.00847297  xend= 0.00312316
EvtNo= 933  length= 0.0083863  xend= 0.000298443
EvtNo= 937  length= 0.00845403  xend= 0.00389289
EvtNo= 950  length= 0.00843045  xend= 0.00285168
EvtNo= 954  length= 0.0084763  xend= 0.000905436
EvtNo= 965  length= 0.00842874  xend= 0.00112399
EvtNo= 966  length= 0.00848528  xend= 0.003883
EvtNo= 970  length= 0.0084922  xend= 0.00349534
EvtNo= 984  length= 0.00843179  xend= 0.00366124
EvtNo= 995  length= 0.00854747  xend= 0.00236942
Run terminated.
Run Summary
  Number of events processed : 1000
  User=4.12s Real=10.31s Sys=2.22s
 ================== run summary =====================

 end of Run TotNbofEvents = 857 for IonC12 with Ekin = 6 MeV
    Track Length in absorber = 8.45568 +- 0.00137474  microns 
    CSDA  Range  in absorber = 7.74892 +- 0.0915186  microns 

    Energy deposit in absorber = 6  MeV 

Average Number electrons per event = 0
Average Number photons per event   = 0

#
/test17/cutGammaEnergy 250 eV
 Cut for secondary gamma = 0.25 keV
/gun/particle proton
/gun/energy 500 keV
/run/beamOn 100

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes     recalculation needed : No 
 Material : Air
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1 eV     e- 4.01595 eV     e+ 4.01595 eV 
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 1     used in the geometry : Yes     recalculation needed : No 
 Material : Water
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1.07536 keV    e- 84.6034 keV    e+ 83.5687 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

====================================================================

### Run 3 start.
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 100
  User=9.88s Real=10.12s Sys=0.2s
 ================== run summary =====================

 end of Run TotNbofEvents = 100 for proton with Ekin = 0.5 MeV
    Track Length in absorber = 9.15247 +- 0.0150483  microns 
    CSDA  Range  in absorber = 9.03455 +- 0.014533  microns 

    Energy deposit in absorber = 0.5  MeV 

### Comparison with the ICRU49 data: 
    Track Length (G4 - ICRU49) = 0.0934701 +- 0.0150483 microns 
    CSDA  Range  (G4 - ICRU49) = 0.165555 +- 0.014533 microns 

Average Number electrons per event = 0.07
Average Number photons per event   = 0.07

#
/test17/cutAugerEnergy 250 eV
 Cut for Auger electrons = 0.25 keV
/run/beamOn 100

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes     recalculation needed : No 
 Material : Air
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1 eV     e- 4.01595 eV     e+ 4.01595 eV 
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 1     used in the geometry : Yes     recalculation needed : No 
 Material : Water
 Range cuts        :  gamma 100 mum    e- 100 mum    e+ 100 mum
 Energy thresholds :  gamma 1.07536 keV    e- 84.6034 keV    e+ 83.5687 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

====================================================================

### Run 4 start.
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 100
  User=10.31s Real=10.62s Sys=0.12s
 ================== run summary =====================

 end of Run TotNbofEvents = 100 for proton with Ekin = 0.5 MeV
    Track Length in absorber = 9.18379 +- 0.0168552  microns 
    CSDA  Range  in absorber = 9.05791 +- 0.017086  microns 

    Energy deposit in absorber = 0.5  MeV 

### Comparison with the ICRU49 data: 
    Track Length (G4 - ICRU49) = 0.124793 +- 0.0168552 microns 
    CSDA  Range  (G4 - ICRU49) = 0.188914 +- 0.017086 microns 

Average Number electrons per event = 19.14
Average Number photons per event   = 0.09

G4 kernel has come to Quit state.
